python/cudf/dataframe/dataframe.py

```diff
@@ -1640,8 +1640,9 @@ def fix_name(name, suffix):
                 rhs_column_idx = rhs_column_idx + 1
 
         if left_index and right_index:
-            df = df.drop(lhs.LEFT_RIGHT_INDEX_NAME)
-            df = df.set_index(lhs.index[df.index.gpu_values])
+            df = df.set_index(lhs.LEFT_RIGHT_INDEX_NAME)
+            df.index.name = None
```

Comment:
Do we need to reset the name of the index to none? Should it retain the name of the left table's index?

Reply:
Right as always sir! :D Pandas will retain the index name if it has one, now we will, too. Also discovered and fixed a bug that index names were being dropped when using `sort_values`.